### PR TITLE
implement count feature as a separate brick

### PR DESCRIPTION
**Pull request by JvnSlv**

### CODE REVIEW

**Comment by StanevPrime** [comment]
@PVGeorgievPrime, please discuss the issues we found with @JvnSlv

File: packages/rx_bloc_cli/lib/src/commands/create_command.dart [hunk 1]
```diff
@@ -46,6 +46,7 @@ class CreateCommand extends Command<int> {
   final _projectNameString = 'project-name';
   final _organisationString = 'organisation';
   final _analyticsString = 'enable-analytics';
+  final _counterString = 'feature_counter';
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/lib/src/commands/create_command.dart:49]: What do you think of the following param name: enable-feature-counter ?

**Review by StanevPrime — CHANGES_REQUESTED** [review]
Please make sure that all counter related code won't be added to the generated project if the feature is disabled. @PVGeorgievPrime can help with this challenge

File: packages/rx_bloc_cli/lib/src/commands/create_command.dart (hunk 1 above)
Comment (JvnSlv) [reply packages/rx_bloc_cli/lib/src/commands/create_command.dart:49]: Sounds better 👍

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/di/{{project_name}}_with_dependencies.dart [hunk 2]
```diff
@@ -66,14 +70,14 @@ class {{project_name.pascalCase()}}WithDependencies extends StatelessWidget {
...
-  List<SingleChildWidget> get _coordinator => [
+  List<SingleChildWidget> get  _coordinator => [
         RxBlocProvider<CoordinatorBlocType>(
           create: (context) => CoordinatorBloc(),
         ),
       ];
 
   {{#analytics}}
-  List<Provider> get _analytics => [
+  List<Provider> get  _analytics => [
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/di/{{project_name}}_with_dependencies.dart:80]: Is the added space in compliance with dart formatter ?

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/repositories/{{#enable_feature_counter}}counter_repository.dart{{/enable_feature_counter}} [hunk 3]
```diff
@@ -1,8 +1,8 @@
 {{> licence.dart }}
 
+import '../../feature_counter/models/count.dart';
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/repositories/{{#enable_feature_counter}}counter_repository.dart{{/enable_feature_counter}}:3]: All files in the base folder should be completely decoupled from the features. With that being said I believe the model should not be moved to the feature but should reside in the base/models folder instead

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/feature_splash/blocs/splash_bloc.dart [hunk 4]
```diff
@@ -64,7 +64,13 @@ class SplashBloc extends $SplashBloc {
       _navigationBloc.events.goToLocation(_redirectLocation!);
     } else {
       await _authService.isAuthenticated()
-          ? _navigationBloc.events.go(const CounterRoute())
+          ? 
+          {{#enable_feature_counter}}
+          _navigationBloc.events.go(const CounterRoute())
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/feature_splash/blocs/splash_bloc.dart:69]: Considering the counter will become an optional feature I believe we can add a new one called dashboard, which will be in the first position in the nav menu as well as the default page, which the logged users will see.

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/test/{{#enable_feature_counter}}feature_counter{{/enable_feature_counter}}/blocs/counter_bloc_test.dart [hunk 5]
```diff
@@ -5,10 +5,10 @@ import 'package:mockito/annotations.dart';
 import 'package:mockito/mockito.dart';
 import 'package:rx_bloc/rx_bloc.dart';
 import 'package:rx_bloc_test/rx_bloc_test.dart';
-import 'package:{{project_name}}/base/models/count.dart';
 import 'package:{{project_name}}/base/models/errors/error_model.dart';
 import 'package:{{project_name}}/base/repositories/counter_repository.dart';
 import 'package:{{project_name}}/feature_counter/blocs/counter_bloc.dart';
+import 'package:{{project_name}}/feature_counter/models/count.dart';
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/test/{{#enable_feature_counter}}feature_counter{{/enable_feature_counter}}/blocs/counter_bloc_test.dart:11]: Same comment as above

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/hooks/post_gen.dart [hunk 6]
```diff
@@ -1,6 +1,23 @@
...
+
 import 'package:mason/mason.dart';
 
-void run(HookContext context) {
+void run(HookContext context) async {
   // Add your post-generation code here
   // (Code executed immediately after the generation step)
+
+  //Builds generated packages
+  if (context.vars['enable_feature_counter'] == false) {
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/hooks/post_gen.dart:10]: Please sync this with @RomanovaPrime

**Review by StanevPrime — CHANGES_REQUESTED** [review]

**Comment by StanevPrime** [comment]
@JvnSlv , be aware that the CI is running on Linux, which means that you should not regenerate the golden files on Mac. You can find more information here - https://github.com/flutter/flutter/issues/36667

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/hooks/post_gen.dart (hunk 6 above)
Comment (StanevPrime) [reply packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/hooks/post_gen.dart:10]: It looks like that the post_gen is not working as expected so the code has been moved to the main command execution - https://github.com/Prime-Holding/rx_bloc/pull/431

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/repositories/{{#enable_feature_counter}}counter_repository.dart{{/enable_feature_counter}} (hunk 3 above)
Comment (JvnSlv) [reply packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/repositories/{{#enable_feature_counter}}counter_repository.dart{{/enable_feature_counter}}]: Done

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/test/{{#enable_feature_counter}}feature_counter{{/enable_feature_counter}}/blocs/counter_bloc_test.dart (hunk 5 above)
Comment (JvnSlv) [reply packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/test/{{#enable_feature_counter}}feature_counter{{/enable_feature_counter}}/blocs/counter_bloc_test.dart]: Done

**Review by StanevPrime — APPROVED** [review]

**Pull request merged**